### PR TITLE
[ENG-879] Parallel Schedule stop "MySQL has gone away errors"

### PR DESCRIPTION
**Pull request by heathdutton**
This code looks a little funky, but it’s following the pattern of the
DBAL connection logic as much as possible. I’m supporting
non-autocommit and rollback-only even though those modes *shouldn’t* be
used in this project. With this the buffer errors do not occur.

### CODE REVIEW

File: Model/Api.php [hunk 1]
```diff
@@ -1780,7 +1782,7 @@ public function kickoffParallelCampaigns(Contact $contact, &$campaignIds = [])
             return $this;
         }
 
-        if (!function_exists('pcntl_fork') || PHP_SAPI !== 'cli') {
+        if (!$allowFork || !function_exists('pcntl_fork') || PHP_SAPI !== 'cli') {
```
Comment (cykonetic) [inline Model/Api.php]: Demorgan strikes again ...

(!($allowFork && function_exists('pcntl_fork') && PHP_SAPI === 'cli'))

File: Model/Api.php [hunk 2]
```diff
@@ -1833,11 +1835,24 @@ public function kickoffParallelCampaigns(Contact $contact, &$campaignIds = [])
...
         $connection = $this->em->getConnection();
-        if ($connection->isConnected()) {
-            if ($connection->isTransactionActive()) {
-                $connection->commit();
+        if ($connection) {
+            while (0 !== $connection->getTransactionNestingLevel()) {
+                // Check for RollBackOnly to avoid exceptions.
+                if (!$connection->isRollbackOnly()) {
+                    // Follow behavior of the private commitAll method, in case there are nested transactions.
+                    if (false === $connection->isAutoCommit() && 1 === $connection->getTransactionNestingLevel()) {
```
Comment (dean-martin) [inline Model/Api.php]: I'm not too familiar with Doctrine, but if it's possible to run `$connection->commit();` before the 
`if (false === $connection->isAutoCommit() && 1 === $connection->getTransactionNestingLevel()) {`
It'd save a line of code, but I'm not sure if that'd make the `if` check invalid.

File: Model/Api.php (hunk 1 above)
Comment (heathdutton) [reply Model/Api.php]: [wut](https://i.imgur.com/EYYSlsv)

File: Model/Api.php (hunk 2 above)
Comment (heathdutton) [reply Model/Api.php]: This is a weird one, I'm trying to pretend to effectively run commitAll which is a private method that I can't call, so this while loop logic is effectively whats nested in there, just using public methods.

Honestly, though, I'm wondering if this is enough... What about flushing/persisting entity manager changes?

**Pull request merged**